Mu2eUtilities/src/GammaPairConversionSpectrum.cc

```diff
@@ -0,0 +1,257 @@
+    if (delta > 1.4) {
+      f1 = 42.038 - 8.29*log(delta + 0.958);
+      f2 = f1;
+    } else {
+      f1 = 42.184 - delta*(7.444 - 1.623*delta);
+      f2 = 41.326 - delta*(5.848 - 0.902*delta);
+    }
+  }
+
+  //From G4ModifiedTsai::SamplePairDirections
+  void GammaPairConversionSpectrum::samplePairDirections(const CLHEP::HepLorentzVector &photon, double electron_ke,
+  double positron_ke, CLHEP::Hep3Vector &electron_dir,CLHEP::Hep3Vector &positron_dir) {
+    CLHEP::Hep3Vector photon_dir(photon.vect());
+    photon_dir.setMag(1.);
+
+    double phi = CLHEP::twopi * _rndFlat->fire();
+    double sinp = std::sin(phi);
+    double cosp = std::cos(phi);
+
+    double u(0.), theta(0.), cost(0.), sint(0.);
+    if(_correlateAngleOverKE) {
+      u = GammaPairConversionSpectrum::sampleThetaU(electron_ke);
+      theta = u*_me/electron_ke;
+      cost = cos(theta);
+    } else
+      cost = GammaPairConversionSpectrum::sampleCosTheta(electron_ke);
+    sint = std::sqrt((1.-cost)*(1.+cost));
+
+    electron_dir.set(sint*cosp, sint*sinp, cost);
+    electron_dir.rotateUz(photon_dir);
+
+    if(_correlateAngleOverKE) {
+      theta = u*_me/positron_ke;
+      cost = cos(theta);
+    } else
+      cost = GammaPairConversionSpectrum::sampleCosTheta(positron_ke);
+    sint = std::sqrt((1.-cost)*(1.+cost));
+
+    positron_dir.set(-sint*cosp, -sint*sinp, cost);
+    positron_dir.rotateUz(photon_dir);
+  }
+
+  //From G4ModifiedTsai::SampleCosTheta(G4double kinEnergy)
+  double GammaPairConversionSpectrum::sampleCosTheta(double ke) {
+    double uMax = 2.*(1. + ke/_me);
+    static const double a1 = 1.6;
+    static const double a2 = a1/3.;
+    static const double border = 0.25;
+    double u(0.);
+    do {
+      double rndnum1 = _rndFlat->fire();
+      const double rndnum2 = _rndFlat->fire();
+      const double uu = -log(rndnum1*rndnum2);
+      rndnum1 = _rndFlat->fire();
+      u = (border > rndnum1) ? uu*a1 : uu*a2;
+    } while(u > uMax);
+    //in GEANT 4.10.p03b used cos(u*me/ke) ~ 1 - (u*me/ke)^2 / 2 = 1 - 2*u^2/ (2 ke/me)^2 ~ 1 - 2*u^2/(2*(1+ke/me))^2
+    return 1. - 2.*u*u/uMax/uMax;
+  }
+
+  //From G4BetheHeitlerModel::SampleSecondaries, used in 4.10.4.p03b
+  double GammaPairConversionSpectrum::sampleThetaU(double ke) {
+    static const double a1 = 1.6;
+    static const double a2 = a1/3.;
+    static const double border = 0.25;
+    // double uMax = M_PI*ke/_me; //if above this angle > pi
+    double rndnum1 = _rndFlat->fire(), rndnum2 = _rndFlat->fire();
+    const double uu = -log(rndnum1*rndnum2);
+    rndnum1 = _rndFlat->fire();
+    const double u = (border > rndnum1) ? uu*a1 : uu*a2;
+    return u;
+  }
+
+  void GammaPairConversionSpectrum::initializeElementData() {
+    for(int z = 1; z < 121; ++z) {
+      double z3 = pow(z, 1./3.);
+      //from G4Element::ComputeCoulombFactor()
+      const double k1 = 0.0083, k2 = 0.20206, k3 = 0.0020, k4 = 0.0369;
+      const double alpha = 1./137.036; //fine structure constant
```

Comment:
alpha is defined in
"CLHEP/Units/PhysicalConstants.h"
please use the value from there instead.

Reply:
Replacing with CLHEP::fine_structure_constant